src/java.net.http/share/classes/jdk/internal/net/http/common/DebugLogger.java

```diff
@@ -41,10 +41,72 @@
  * format string. MessageFormat-like formatting is enabled by the presence of
  * "{0" or "{1".
  *
- * <p> See {@link Utils#getDebugLogger(Supplier, boolean)} and
- * {@link Utils#getHpackLogger(Supplier, boolean)}.
+ * <p> See {@link Utils#getDebugLogger(Supplier)} and
+ * {@link Utils#getHpackLogger(Supplier)}.
  */
 final class DebugLogger implements Logger {
+
+    /**
+     * A DebugLogger configuration is composed of three levels.
+     * The three levels can be configured independently, but are
+     * typically either Level.ALL or Level.OFF.
+     *
+     * @param outLevel the level above which messages will be directly
+     *                printed to {@link System#out}
+     * @param errLevel the level above which messages will be directly
+     *                printed to {@link System#err}
+     * @param logLevel the level above which messages will be forwarded
+     *               to an underlying {@link System.Logger}
+     */
+    public record LoggerConfig(Level outLevel, Level errLevel, Level logLevel) {
+        public LoggerConfig {
+            Objects.requireNonNull(outLevel);
+            Objects.requireNonNull(errLevel);
+            Objects.requireNonNull(logLevel);
+        }
+
+        // true if at least on of the three levels is not Level.OFF
+        public boolean on() {
+            return minSeverity() <= Level.OFF.getSeverity();
+        }
+        // The minimal severity of the three level. Messages below
+        // that severity will not be logged anywhere.
+        public int minSeverity() {
+            return Math.min(outLevel.getSeverity(),
+                    Math.min(errLevel.getSeverity(), logLevel.getSeverity()));
+        }
+        // Whether the given level can be logged with the given logger
+        public boolean levelEnabledFor(Level level, System.Logger logger) {
+            if (level == Level.OFF) return false;
+            int severity = level.getSeverity();
+            if (severity >= errLevel.getSeverity()) return true;
+            if (severity >= outLevel.getSeverity()) return true;
+            if (severity >= logLevel.getSeverity()) return logger.isLoggable(level);
+            return false;
+        }
+        // The same configuration, but with the given {@link #errLevel}
+        public LoggerConfig withErrLevel(Level errLevel) {
+            return new LoggerConfig(outLevel, errLevel, logLevel);
+        }
+        // The same configuration, but with the given {@link #outLevel}
+        public LoggerConfig withOutLevel(Level outLevel) {
+            return new LoggerConfig(outLevel, errLevel, logLevel);
+        }
+        // The same configuration, but with the given {@link #logLevel}
+        public LoggerConfig withLogLevel(Level logLevel) {
+            return new LoggerConfig(outLevel, errLevel, logLevel);
+        }
```

Comment:
To avoid shadowing, do you think we could rename the method params of these 3 methods to `newErrLevel`, `newOutLevel` and `newLogLevel`? If you prefer, the current form, that's fine too.